website/docs/index.html.markdown

```diff
@@ -373,6 +373,76 @@ The following arguments are used to configure the VMware Cloud Director Provider
 * `import_separator` - (Optional; *v2.5+*) The string to be used as separator with `terraform import`. By default
   it is a dot (`.`).
 
+* `ignore_metadata` - (Optional; *v3.10+*) Use one or more of these blocks to ignore specific metadata entries.
+  See ["Ignore Metadata"](#ignore-metadata) for more details.
+
+## Ignore metadata
+
+One or more `ignore_metadata` blocks can be optionally set in the provider configuration, which will allow to ignore specific `metadata_entry`
+items during `plan`, `apply` and `destroy`. This is useful, for example, to avoid removing metadata entries that were created
+by an external actor and not by this provider.
+
+~> Note that this feature is only considered when using the `metadata_entry` argument in the resources and data sources that support
+it. In other words, to ignore metadata when you are using the deprecated `metadata` argument, please use the native Terraform `lifecycle` block.
+
+-> To avoid side effects, you should only use this feature to deal with metadata not managed by this provider.
+It is encouraged to avoid ignoring metadata that was created by a Terraform resource, as you could get unwanted updates or inconsistencies
+in the state.
+
+The available sub-attributes for `ignore_metadata` are:
+
+* `object_type` - (Optional) Specifies the type of the object which metadata needs to be ignored. The object types must be declared
+  as defined in the metadata endpoints of the [VCD API documentation](https://developer.vmware.com/apis/1601/vmware-cloud-director). For example,
+  `catalog`, `media`, `disk`, `org`, `catalogItem`, `vAppTemplate`, `vdc`, `network`, `providervdc`, `vApp` (note this one is used
+  for both vApps and VMs), etc.
```

Comment:
Thoughts:
1. The resource list needs to be exact and specific. It shouldn't be "for example". The reason is that we need to tell users which resources this is actual for.
2. I agree that it would be easier if these types mapped to Terraform resources. It would also be more managable and readable in HCL.

Reply:
I agree with both suggestions, if possible we should try and use resource names instead of API objects, also specificity is very important, or every resource should have that information clearly stated in the documentation.

Reply:
I had the same thought, as it simplifies the user experience, but maintaining this map ("resource name"->"metadata endpoint") is what pulled me back.
I'll do the changes and we can check.

Reply:
Maybe we can utilize (or extend) resource list that @dataclouder had implemented earlier for holding this mapping? If the resource list already has some maintainable list/map, it's better to have everything in one place.

Reply:
Done, I created a small function that handles the mapping.

Reply:
I like the way it became easy to use with the `resource_type` field and all the familiar resource names!

@dataclouder are you OK with this being a separate metadata-centric function or should this be integrated into the `vcd_resource_list`/`vcd_resource_schema` framework?

Reply:
I have attempted this new approach manually - I think this is much more convenient for users - thanks!

One observation to share - I think we should validate the `resource_type` field in `provider`. I can make a typo or put `sfSDFSDFSD` and then `resource_type` would be ignored (filtering applied to all resources/data source)

Reply:
> @dataclouder are you OK with this being a separate metadata-centric function or should this be integrated into the vcd_resource_list/vcd_resource_schema framework?
I don't see how this could be integrated into `vcd_resource_list`. They have different purposes

Reply:
@Didainius Yes, that's a good idea. I think @adezxc also had [a similar one](https://github.com/vmware/terraform-provider-vcd/pull/1057#discussion_r1253302365), so you both have a point.
I have added a ValidateFunc to `resource_type`

Reply:
I am closing this issue as I think we have it sorted now.